### PR TITLE
[MRG] relocate cDBG files to cDBG directory

### LINKED ISSUE #347: move more cdbg stuff into the new cdbg directory?
things like the gxt file, and the k-mer indices, probably belong in the cdbg directory (e.g. `dory_k21`) rather than in the catlas directory (`dory_k21_r1`)

### LINKED ISSUE #352: figure out how to test `spacegraphcats/snakemake.py`
and/or reconsider having it at all ;)

(I wonder if it's possible to integrate it into the Snakefile?)

### LINKED ISSUE #374: move and/or rename various indices.
misc cleanup to make things make more sense and be consistent:

- `twofoo-short/twofoo-short.reads.bgz` should be just `twofoo-short/reads.bgz` I think
- `twofoo-short_k31/bcalm.unitigs.fa` should be deleted after indexing is run
- `twofoo-short_k31_r1/bcalm.twofoo-short_k31_r1.log.txt` probably belongs as `twofoo-short_k31/bcalm.log.txt`
- `twofoo-short_k31_r1/contigs.*` probably all belong in `twofoo-short_k31/` instead of `twofoo-short_k31_r1/`

### PR DESCRIPTION
**Pull request by ctb**
Many contigs/cDBG files were located in the catlas directory before, which meant they were being unnecessarily rebuilt for each new catlas radius.

This PR relocates the following files into the cDBG directory (e.g. `dory_k21`) from the catlas directory (e.g. `dory_k21_r1`):
* `cdbg.txt`
* `contigs.mphf`
* `contigs.indices`
* `contigs.info.csv`
* `contigs.sig`,
* `reads.bgz.index`

This PR also renames the bcalm log file to `bcalm.log.txt`.

Along the way, I also fixed the target generation code in `spacegraphcats/snakemake.py` and added it into the tests, viz #352.

**Note:** this changes the names / location of many files! Will require cDBG and catlas rebuilds.

Fixes https://github.com/spacegraphcats/spacegraphcats/issues/347
Fixes https://github.com/spacegraphcats/spacegraphcats/issues/374
Fixes https://github.com/spacegraphcats/spacegraphcats/issues/352

### CODE REVIEW

**Comment by ctb** [comment]
ready for review & merge, @taylorreiter

File: spacegraphcats/snakemake.py [hunk 1]
```diff
@@ -5,29 +5,31 @@
...
 
     z = []
-    z.append(os.path.join(catlas_base, "bcalm.{}.k31.unitigs.fa".format(catlas_base)))
-    z.append(os.path.join(dirname, "catlas.csv"))
-    z.append(os.path.join(dirname, "cdbg.gxt"))
-    z.append(os.path.join(dirname, "contigs.fa.gz.indices"))
-    z.append(os.path.join(dirname, "contigs.fa.gz.info.csv"))
-    z.append(os.path.join(dirname, "contigs.fa.gz.mphf"))
-    z.append(os.path.join(dirname, "first_doms.txt"))
+    z.append(os.path.join(cdbg_dir, "bcalm.unitigs.db")),
```
Comment (taylorreiter) [inline spacegraphcats/snakemake.py:18]: removed the hardcoded `k31` in the name bc the k size is flexible from the conf file?

**Comment by ctb** [comment]
I am optimistic that the tests will pass this time, so, pls review when you have time, @taylorreiter :)

File: spacegraphcats/snakemake.py [hunk 2]
```diff
@@ -54,10 +56,10 @@ def catlas_search(conf_file, cdbg_only=False, suffix=""):
     return z
 
 
-def catlas_extract(conf_file, cdbg_only=False, suffix=""):
-    "Produce the list of files output by 'spacegraphcats <config> extract_contigs extract_reads"
+def catlas_extract_contigs(conf_file, cdbg_only=False, suffix=""):
```
Comment (taylorreiter) [inline spacegraphcats/snakemake.py:59]: Nice. I like that these are split now.

File: spacegraphcats/snakemake.py [hunk 3]
```diff
@@ -74,16 +76,40 @@ def catlas_extract(conf_file, cdbg_only=False, suffix=""):
...
+    for x in filenames:
+        x = os.path.basename(x)
+        z.append(os.path.join(dirname, "{}.cdbg_ids.reads.gz".format(x)))
+
+    return z
+
+
 def catlas_search_input(conf_file):
-    "Produce the list of files output by 'spacegraphcats <config> search"
+    "Produce the list of files required by 'spacegraphcats <config> search"
```
Comment (taylorreiter) [inline spacegraphcats/snakemake.py:110]: nice subtle change in wording

**Review by taylorreiter — APPROVED** [review]
This looks great! I've always been confused by the placement of the cdbg in the `r1` folder, big organizational improvement! Thank you!

File: spacegraphcats/snakemake.py (hunk 1 above)
Comment (ctb) [reply spacegraphcats/snakemake.py:18]: ooh, good catch!

Comment (ctb) [reply spacegraphcats/snakemake.py:18]: oh. I misunderstood.

Umm, the ksize is already in the directory name here.

**Pull request merged**